### ISSUE TITLE
Implementing the EvaluateInterleavedChunks Evaluator

### ISSUE THREAD

**Issue by gilbertoolimpio** [scikit-multiflow/scikit-multiflow#77]
Hello guys!

I'm working with the MOA and there is the **EvaluateInterleavedChunks** evaluator. From what I read [here](https://www.cms.waikato.ac.nz/~abifet/book/chapter_6.html) it is "Also like interleaved test-then-train, but with chunks of data in sequence. Chunks of different sizes may need to be considered "

Are there any plans for the implementation of this evaluator?

Regards!

**Comment by jacobmontiel** [comment]
Hello @gilbertoolimpio,

The `EvaluateInterleavedChunks` is not currently in our plans but we welcome collaboration from the community. We can provide support in case you or somebody else is interested in implementing it.

**Comment by gilbertoolimpio** [comment]
I'm extremely interested in implementing it! I'll get information to start this work! Thank you!

**Comment by jacobmontiel** [comment]
Awesome, let me know if I can be of help.

**Comment by gilbertoolimpio** [comment]
Hello @jacobmontiel!

I started the implementation, however for my tests I'm using a csv file that I was already using on the MOA (there in the arff format). This file has some nominal categorical data (the label is an example) and in the application it throws an exception saying: `File contains text data`. 

My question is: can't this version work with nominal categorical data?

Tks!

**Comment by jacobmontiel** [comment]
Hello @gilbertoolimpio ,

`scikit-multiflow` is designed under the assumption that all data (numerical and nominal) corresponds to a number-type (int, float). This is why the `FileStream` raises the error.

A workaround is to pre-process the data in Python so you only have number-types (no text) and then you can use the `DataStream` to setup the stream.

Another option is to use a different tool (e.g. Weka) to convert the data in your file to numbers and then load the converted dataset with `FileStream`.

Let me know if this clarifies things.

**Comment by gilbertoolimpio** [comment]
> Hello @gilbertoolimpio ,
> 
> `scikit-multiflow` is designed under the assumption that all data (numerical and nominal) corresponds to a number-type (int, float). This is why the `FileStream` raises the error.
> 
> A workaround is to pre-process the data in Python so you only have number-types (no text) and then you can use the `DataStream` to setup the stream.
> 
> Another option is to use a different tool (e.g. Weka) to convert the data in your file to numbers and then load the converted dataset with `FileStream`.
> 
> Let me know if this clarifies things.

Oh! This helped me a lot! From what I saw in the MOA code it does the process of rendering the numerical data transparently to the user! Thank you again

**Comment by jacobmontiel** [comment]
The rationale for this difference is three-folded:

1. We reduce the overhead from handling (converting) data types in the background
2. We allow the user more flexibility in terms of having control on the data.
 3. Different to MOA, `scikit-multiflow` is designed to work alongside other tools such as `scikit-learn`, `pandas`, etc.

**Comment by gilbertoolimpio** [comment]
@jacobmontiel 

I need your help!
By analyzing the code of the `EvaluatedPrequential` class, I noticed that there is a parameter called `batch_size` where the default is `1`. However, when it is used with a value other than `1`, it behaves much like `Interleaved Chunks`. Because it selects a number of instances to first test and then train (running the test-than-train). Is my interpretation wrong? If I'm wrong, is there any hint that I can continue with the implementation?
Could you help me with this question?
Thanks in advanced! =)

**Comment by jacobmontiel** [comment]
Hi @gilbertoolimpio 

If `batch_size=1`, then train and test are performed instance by instance. If `batch_size = n` then `n` samples are queried from the stream and used for testing and training. So they work on equal "chunks" of data.

I am not very familiar with `EvaluateInterleavedChunks` but from this

> Chunks of different sizes may need to be considered 

it is my understanding that you could set different sizes for train and test "chunks"

**Comment by jacobmontiel** [comment]
There have been no updates on this for some time. @gilbertoolimpio is it ok if we label this issue as *help wanted* in case somebody is interested in implementing it?

**Comment by jacobmontiel** [comment]
Closing this issue as there are no updates/changes.
